app/domain/authentication/update_authenticator_config.rb

```diff
@@ -0,0 +1,60 @@
+require 'command_class'
+
+module Authentication
+
+  UpdateAuthenticatorConfig = CommandClass.new(
+    dependencies: {
+      webservice_class: ::Authentication::Webservice,
+      resource_class: ::Resource,
+      authenticator_config_class: ::AuthenticatorConfig
+    },
+    inputs: %i(account authenticator service_id enabled current_user)
+  ) do
+    
+    def call
+      validate_resource_visible
+      validate_resource_is_authenticator
+      validate_resource_writable
+
+      update_authenticator_config
+    end
+
+    private
+
+    def validate_resource_visible
+      raise Exceptions::RecordNotFound, resource_id \
+        unless resource.visible_to?(@current_user)
+    end
+
+    def validate_resource_is_authenticator
+      raise Errors::Authentication::AuthenticatorNotFound, resource_id \
+        unless Authentication::InstalledAuthenticators.configured_authenticators
```

Comment:
`Authentication::InstalledAuthenticators.configured_authenticators` can be a dependency